webots_ros2_driver/webots_ros2_driver/webots_launcher.py

```diff
@@ -82,3 +124,57 @@ def __init__(self, output='screen', world=None, gui=True, mode='realtime', strea
             ],
             **kwargs
         )
+
+    def execute(self, context: LaunchContext):
+        # Check if the user wants to convert URDF files into robots
+        if self.__robots:
+            worldPath = self.__world.perform(context)
+            if not worldPath:
+                sys.exit('World file not specified (has to be specified with world=path/to/my/world.wbt')
+
+            worldCopy = worldPath[:-4] + '_with_URDF_robot.wbt'
+            shutil.copyfile(worldPath, worldCopy)
+            worldName=context.launch_configurations['world']
+            context.launch_configurations['world']=worldName[:-4] + '_with_URDF_robot.wbt'
+
+            for robot in self.__robots:
+                file_input = robot.get('urdf_location')
+                robot_name = robot.get('name')
+                robot_translation = robot.get('translation')
+                robot_rotation = robot.get('rotation')
+
+                if not file_input:
+                    sys.exit('URDF file not specified (has to be specified with \'urdf_location\': \'path/to/my/robotUrdf.urdf\'')
+                if not robot_name:
+                    sys.stderr.write('Robot name not specified (should be specified if more than one robot is present with \'name\': \'robotName\'\n')
+
+                convert2urdf(inFile=file_input, worldFile=worldCopy, robotName=robot_name, initTranslation=robot_translation, initRotation=robot_rotation)
```

Comment:
The added `worldFile` argument here breaks the "abstraction". We should use the urdf2webots tools as is (with small changes). Instead, I would change the `convert2urdf` function to return a URDF file as a string, or we can use it as is but write the output to a temp file.

Reply:
So you mean that `convert2urdf` will still convert URDF to PROTO but also either append a robot to a world file or return the robot in VRML as a string?

Reply:
`convert2urdf` should never append to a world file, it should be handled outside of the `webots2urdf` package.